Okta.Xamarin/Okta.Xamarin/ViewModels/DiagnosticsViewModel.cs

```diff
@@ -13,23 +16,82 @@ public DiagnosticsViewModel(DiagnosticsPage diagnosticsPage)
         {
             this.Page = diagnosticsPage;
             this.StateManager = OktaContext.Current.StateManager;
+            OktaContext.Current.GetUserCompleted += (sender, args) =>
+            {
+				this.userInfo = args.UserInfo;
+				this.ClaimsPrincipal = args.UserInfo as ClaimsPrincipal; // user info may be Dictionary<string, object> or ClaimsPrincipal or generic type; will be null if incorrect 'as' cast
+			};
         }
 
         protected DiagnosticsPage Page { get; }
 
         public RevokeAccessTokenCommand RevokeAccessTokenCommand => new RevokeAccessTokenCommand();
 
-        OktaStateManager stateManager;
+        public GetClaimsPrincipalCommand GetClaimsPrincipalCommand => new GetClaimsPrincipalCommand();
+
+		public GetUserCommand GetUserCommand => new GetUserCommand();
+
+        IOktaStateManager stateManager;
 
-        public OktaStateManager StateManager
+        public IOktaStateManager StateManager
         {
             get { return stateManager; }
-            
+
             set
             {
                 stateManager = value;
                 OnPropertyChanged(nameof(StateManager));
             }
         }
+
+        object userInfo;
+
+        public object UserInfo
```

Comment:
If we have a `UserInfo` class, why is this an object?

Reply:
This is so the type this property is set to is not restricted; I did this to simplify some of the refactoring that may have to be done to account for custom claims work that still needs to be done on OKTA-371439.